test/e2e/automated_cleaning_test.go

```diff
@@ -0,0 +1,247 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/dypflying/go-qcow2lib/qcow2"
+	metal3api "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"libvirt.org/go/libvirt"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/deprecated/v1beta1/patch"
+)
+
+var _ = Describe("automated-cleaning", Label("required", "automated-cleaning"), func() {
+	var (
+		specName      = "automated-cleaning"
+		secretName    = "bmc-credentials"
+		namespace     *corev1.Namespace
+		cancelWatches context.CancelFunc
+	)
+
+	BeforeEach(func() {
+		namespace, cancelWatches = framework.CreateNamespaceAndWatchEvents(ctx, framework.CreateNamespaceAndWatchEventsInput{
+			Creator:   clusterProxy.GetClient(),
+			ClientSet: clusterProxy.GetClientSet(),
+			Name:      fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
+			LogFolder: artifactFolder,
+		})
+	})
+
+	It("should perform automated cleaning when enabled", func() {
+		By("Creating a secret with BMH credentials")
+		bmcCredentialsData := map[string]string{
+			"username": bmc.User,
+			"password": bmc.Password,
+		}
+		CreateSecret(ctx, clusterProxy.GetClient(), namespace.Name, secretName, bmcCredentialsData)
+
+		By("Creating a BMH")
+		bmh := metal3api.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      specName,
+				Namespace: namespace.Name,
+				Annotations: map[string]string{
+					metal3api.InspectAnnotationPrefix:   "disabled",
+					metal3api.HardwareDetailsAnnotation: hardwareDetails,
+				},
+			},
+			Spec: metal3api.BareMetalHostSpec{
+				Online: true,
+				BMC: metal3api.BMCDetails{
+					Address:         bmc.Address,
+					CredentialsName: "bmc-credentials",
+				},
+				BootMode:       metal3api.Legacy,
+				BootMACAddress: bmc.BootMacAddress,
+			},
+		}
+		err := clusterProxy.GetClient().Create(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the BMH to become available")
+		WaitForBmhInProvisioningState(ctx, WaitForBmhInProvisioningStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateAvailable,
+		}, e2eConfig.GetIntervals(specName, "wait-available")...)
+
+		// Only create second disk for environments with Ironic (e.g. not fixture)
+		if e2eConfig.GetVariable("DEPLOY_IRONIC") == "true" {
+			By("Creating second disk for the VM")
+			if err = CreateAndAttachSecondDisk(ctx, bmc.Name); err != nil {
+				Fail(fmt.Sprintf("Failed to create and attach second disk: %v", err))
+			}
+		} else {
+			Logf("Skipping second disk creation because Ironic is not deployed")
+		}
```

Comment:
This is clever, and works well in CI with VMs, however, we cannot create extra disks like this on real hardware 😅 and I would really like these tests to work for that scenario also.
Could you instead move this so that we do it with the vbmctl before the test starts? I get that it will be "wasteful" to add disks to all, even when this test is not going to run, but I don't think it is a big deal.